thorlcr/master/thmastermain.cpp

```diff
@@ -429,11 +423,7 @@ class CRegistryServer : public CSimpleInterface
                 {
                     Owned<IException> e = deserializeException(msg);
                     EXCLOG(e, "Registration error");
-                    if (TE_FailedToRegisterSlave == e->errorCode())
-                    {
-                        setExitCode(0); // to avoid thor auto-recycling
```

Comment:
NB: this was not needed before, the deliberate suppression of Thor restarts in bare-metal at this point, happens [deliberatly] because it hasn't got as far as creating the sentinel file (see init_thor script for non-restart when sentinel missing)